cmd/dep/govendor_importer.go

```diff
@@ -0,0 +1,219 @@
+	return "govendor"
+}
+
+func (g *govendorImporter) HasDepMetadata(dir string) bool {
+	y := filepath.Join(dir, govendorDir, govendorName)
+	if _, err := os.Stat(y); err != nil {
+		return false
+	}
+	return true
+}
+
+func (g *govendorImporter) Import(dir string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	err := g.load(dir)
+	if err != nil {
+		return nil, nil, err
+	}
+	return g.convert(pr)
+}
+
+func (g *govendorImporter) load(projectDir string) error {
+	g.logger.Println("Detected govendor configuration files...")
+	v := filepath.Join(projectDir, govendorDir, govendorName)
+	if g.verbose {
+		g.logger.Printf("  Loading %s", v)
+	}
+	vb, err := ioutil.ReadFile(v)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to read %s", v)
+	}
+	err = json.Unmarshal(vb, &g.file)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to parse %s", v)
+	}
+	return nil
+}
+
+func (g *govendorImporter) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	g.logger.Println("Converting from vendor.json...")
+
+	manifest := &dep.Manifest{
+		Constraints: make(gps.ProjectConstraints),
+	}
+
+	if len(g.file.Ignore) > 0 {
+		// Govendor has three use cases here
+		// 1. 'test' - special case for ignoring test files
+		// 2. build tags - any string without a slash (/) in it
+		// 3. path and path prefix - any string with a slash (/) in it.
+		//   The path case could be a full path or just a prefix.
+		// Dep doesn't support build tags right now: https://github.com/golang/dep/issues/120
+		for _, i := range strings.Split(g.file.Ignore, " ") {
+			if !strings.Contains(i, "/") {
+				g.logger.Printf("Warning: Not able to convert ignoring of build tag '%v'", i)
+				continue
+			}
+			_, err := g.sm.DeduceProjectRoot(i)
+			if err == nil {
+				manifest.Ignored = append(manifest.Ignored, i)
+			} else {
+				g.logger.Printf("Warning: Not able to convert ignoring of build tag '%v'\n", i)
+			}
+		}
+	}
+
+	lock := &dep.Lock{}
+	for _, pkg := range g.file.Package {
+		// Path must not be empty
+		if pkg.Path == "" {
+			err := errors.New("Invalid govendor configuration, Path is required")
+			return nil, nil, err
+		}
+
+		// Obtain ProjectRoot. Required for avoiding sub-package imports.
+		// Use Path instead of Origin since we are trying to group by project here
+		pr, err := g.sm.DeduceProjectRoot(pkg.Path)
+		if err != nil {
+			return nil, nil, err
+		}
+		pkg.Path = string(pr)
+
+		// Check if it already existing in locked projects
+		if projectExistsInLock(lock, pkg.Path) {
+			continue
+		}
+
+		// Revision must not be empty
+		if pkg.Revision == "" {
+			err := errors.New("Invalid govendor configuration, Revision is required")
+			return nil, nil, err
+		}
+
+		if pkg.Version == "" {
+			// When no version is specified try to get the corresponding version
+			pi := gps.ProjectIdentifier{
+				ProjectRoot: gps.ProjectRoot(pkg.Path),
+			}
+			if pkg.Origin != "" {
+				pi.Source = pkg.Origin
```

Comment:
Since we know that any `origin` with a vendor segment is not going to work with dep, how about we check for those, and print a warning when we find an unsupported origin that it was ignored.